### ISSUE TITLE
Cannot access publicURL on a Media field that allows multiple images

### ISSUE THREAD

**Issue by denisgoryaynov** [strapi/gatsby-source-strapi#45]
Currently only Media fields that allow only one image have the publicURL property, so while it's possible to query like this: 
```
{
   allStrapiContent{
      edges{
         node{
            singleImage{
               publicURL
            }
         }
      }
   }
}
```

this results in an error if the field allows multiple images:
```
{
   allStrapiContent{
      edges{
         node{
            multipleImages{
               publicURL
            }
         }
      }
   }
}
```

I've checked the code inside `gatsby-strapi-source/src/normalize.js` and the function `extractFields` launches a useless recursion to normalize images contained inside arrays because the field passed as the 'item' argument is the image object itself, for example: 
```
{
   /* other properties */
   url: 'image url as string',
   mime: 'image mime type as string'
}
```

and later iterates through the objects checking every key with `field.hasOwnProperty('mime')` which is always false since `mime.mime` (or any other field since they are all strings) doesn't exist and because of this doesn't create any node nor downloads the image to be available to use in gatsby-image. I've tweaked the function a bit but the only result I was able to come up with involved creating a `localFile___NODE` and accessing it as:
```
{
   allStrapiContent{
      edges{
         node{
            multipleImages{
               localFile{
                  publicURL
               }
            }
         }
      }
   }
}
```

which I find a bit ugly and incoherent with the syntax mentioned above as the first example.

**Comment by Flaex** [comment]
Hi @denisgoryaynov,
I'm developing a personal project using Strapi and Gatsby and I'm getting through this situation. I ended up [not using gatsby-image](https://www.dropbox.com/s/v58ts1jpuwq8owz/code-1.png?dl=0) , mapping the array images inside a <img> tag and writing manually the Strapi url of the image, obviously this not a good solution since this will not work on the production environment.

It makes no sense using gatsby-images just for single images and leave array images out because of this. I would like to know where you change your code to be able to create the localFile node, thanks in advance!

**Comment by denisgoryaynov** [comment]
Hi, I solved the problem making a fork of this project and changing the code inside src/normalize.js, you should see it in my account. It works and I've been using it on some projects at the company I work, but I still think that the localFile node is an ugly solution but currently that's the best I could come up with and I'm open to other ideas/improvements

**Comment by Flaex** [comment]
To be honest, I'm a newbie in programming. This is my first attempt on building a JAMStack web app so I think I will be learning a lot from you and the community. It is very surprising there is no support for multi-images objects on Gatsby since this is a pretty common feature. 

It could be not the most elegant solution but is solving a situation that would be a common place for many starters on Gatsby and Strapi. I really appreciate your workaround and your quick response.
Many thanks!

**Comment by Flaex** [comment]
Every time I try to `gatsby develop` after installing your gatsby-source-strapi repo all the Strapi's Content Types are inaccessible by GraphQL. This is the message: 

```console
TypeError: Cannot read property 'allStrapiArticle' of undefined

- gatsby-node.js:38 makeRequest.then.result
    e:/wamp/www/@flaex_/app/gatsby-node.js:38:17
```

*gatsby-node.js:38*
````
result.data.allStrapiArticle.edges.forEach(({ node }) => {
````

Then all the GraphQL request fail e.g.
```console
error GraphQL Error Encountered 11 error(s):                                                                            
- Unknown field 'allStrapiLink' on type 'Query'. Source: document `FooterTemplate` file: `GraphQL request`              
  GraphQL request (3:9)
  2:       query FooterTemplate {
  3:         allStrapiLink {
             ^
  4:           edges {
````


There is something I'm missing here?

**Comment by denisgoryaynov** [comment]
Hi, try deleting your cache using `rm -rf .cache` inside your project. Also the master branch of my fork is missing the build files required for this library to work (I don't want to pollute the main branch). I made a second branch called "build" with these files, so if you are using npm just add #build at the end of the repo url inside your package.json and run `npm i` to update the files

**Comment by Flaex** [comment]
Hi @denisgoryaynov. I followed your instructions and now all is working as expected.
When I was searching on the web for alternatives to solve this problem, I found this [helper function](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-filesystem#createremotefilenode) in the gatsby-source-filesystem plugin. Basically is downloading the file and making available in the GraphQL schema. As far I can see, it is adding more complexity to this situation for non experienced devs like me. I'll sitck with your solution, thanks again for your help!

**Comment by aramkoukia** [comment]
any updates on this?

**Comment by denisgoryaynov** [comment]
I opened a pull request to fix this issue #90

**Comment by beve** [comment]
Works great for me. Thank you denisgoryaynov.

Please strapi team, merge ! :)
